addon/poll-task.js

```diff
@@ -158,15 +158,32 @@ export function pollTask(obj, taskOrName, token = getNextToken()) {
    @param { String } token the token for the pollTask to be cleared
    @public
    */
-export function cancelPoll(token) {
+export function cancelPoll(obj, token) {
+  // TODO: remove this older API on next major version
+  if (token === undefined) {
+    deprecate(
+      'ember-lifeline cancelPoll called without an object. New syntax is cancelPoll(obj, cancelId) and avoids a memory leak.',
+      true,
+      {
+        id: 'ember-lifeline-cancel-poll-without-object',
+        until: '4.0.0',
+      }
+    );
+    token = obj;
+    obj = null;
+  }
   delete queuedPollTasks[token];
+  if (obj) {
```

Comment:
I'd prefer this to become an `else` to the `if (token === undefined)` conditional above